src/symbol/projection.js

```diff
@@ -609,43 +583,43 @@ function placeGlyphAlongLine(
             return null;
 
         prev = current;
-        pathVertices.push(current);
-        if (returnPathInTileCoords) tilePath.push(currentVertex || previousTilePoint());
+        prevVertex = currentVertex;
+
+        pathVertices.push(prev);
+        if (returnPathInTileCoords) tilePath.push(prevVertex);
 
+        currentVertex = new Point(lineVertexArray.getx(currentIndex), lineVertexArray.gety(currentIndex));
         current = projectionCache[currentIndex];
-        if (current === undefined) {
-            currentVertex = new Point(lineVertexArray.getx(currentIndex), lineVertexArray.gety(currentIndex));
+        if (!current) {
             const projection = elevatePointAndProject(currentVertex, tileID.canonical, labelPlaneMatrix, reprojection, getElevation);
-            if (projection.signedDistanceFromCamera > 0) {
-                current = projectionCache[currentIndex] = projection.point;
+            if (projection[3] > 0) {
+                current = projectionCache[currentIndex] = projection;
             } else {
                 // The vertex is behind the plane of the camera, so we can't project it
                 // Instead, we'll create a vertex along the line that's far enough to include the glyph
                 // Don't cache because the new vertex might not be far enough out for future glyphs on the same segment
                 current = getTruncatedLineSegment();
             }
-        } else {
-            currentVertex = null; // null stale data
```

Comment:
To confirm, this was unnecessary as it was recreated anyway slightly further down (line 635)?

Reply:
This whole loop had a pretty confusing flow so it's tricky to explain, but the gist is that if `current` value wasn't found in cache, both `current` and `currentVertex` would be computed, and then the latter could be saved to be used as "previous currentVertex" value in the next iteration, but if it was cached, then the previous value had to be recalculated using `previousTilePoint()`. 

So, trying to be do some complicated caching logic to avoid a few calculations likely inadvertedly made it worse while also making the code much harder to read.